### DISCUSSION TITLE
Mitsuba3 Cuda in Ubuntu 24.04 // Optix // libnvoptix.so.1 Could Not be Loaded Fix

### DISCUSSION THREAD

**Discussion by Disgruntoad** [mitsuba-renderer/mitsuba3#1252] [General]
![Screenshot from 2024-07-30 16-58-35](https://github.com/user-attachments/assets/3ea23e5d-d5f7-4609-b44d-386b218c8403)

`jit_optix_api_init(): libnvoptix.so.1 could not be loaded -- disabling OptiX backend! Set DRJIT_LIBOPTIX_PATH environment variable to specify its path....`

Mainly just posting this in case anyone else has this problem with using nvidia's optix  (ray tracing) library. Took me a while to track down the solution on an obscure forum post from a few years ago. I'm not sure if this missing package is installed if you automatically install the drivers. I manually installed nvidia-dkms-560 and had this issue.

You need to install libnvidia-gl-[gpu driver version], eg:

`sudo apt install libnvidia-gl-560`

I also read that you need the display version of the gpu drivers (not 'headless' or 'server' versions). I don't believe that matters. I moved from headless to display drivers on my machine and neither loaded the libnvoptix.so.1 file until I installed the libnvidia-gl library.

**Comment by merlinND** [comment]
Hello @Disgruntoad,

Thanks for sharing your solution.
When installing the NVIDIA drivers in the officially supported ways (e.g. runfile or the CUDA apt package), I don't think that manually installing `libnvidia-gl-*` is needed. Still, good to know!

**Reply by Disgruntoad** [reply]
I did install nvidia drivers with 

`sudo apt install nvidia-dkms-560`

which is, as far as I know, officially supported and one reason I was surprised to have this issue.

**Reply by merlinND** [reply]
I believe the package to install is `nvidia-driver-560`. `nvidia-dkms-*` is just one component.
